Controller/Admin/TransactionTabPostProcessing.php

```diff
@@ -0,0 +1,180 @@
+
+    protected $_i18n = null;
+
+    protected $_config = null;
+
+    /**
+     * TransactionTabPostProcessing constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->_i18n = Registry::getLang();
+        $this->_config = Registry::getConfig();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return string
+     */
+    public function render(): string
+    {
+        $template = parent::render();
+        $transaction = $this->_aViewData['transaction'];
+        $actions = $this->_getPostProcessingActions($transaction);
+        $alert = [];
+
+        try {
+            $this->validateRequest($transaction);
+
+            // if an action was requested, run its callback
+            foreach ($actions as $key => $action) {
+                if ($this->_config->getRequestParameter($key)) {
+                    $alert['message'] = call_user_func([$this, $action['callback']], $transaction);
+                    $alert['type'] = 'success';
+
+                    break;
+                }
+            }
+        } catch (Exception $e) {
+            $alert['message'] = $e->getMessage();
+            $alert['type'] = 'error';
+        }
+
+        $this->_aViewData['actions'] = $actions;
+        $this->_aViewData['alert'] = $alert;
+        $this->_aViewData['currency'] = $transaction->wdoxidee_ordertransactions__wdoxidee_currency->value;
+        $this->_aViewData['requestParameters'] = $this->getRequestParameters();
+
+        return $template;
+    }
+
+    /**
+     * Returns an array of current request parameters.
+     *
+     * @return array
+     */
+    public function getRequestParameters(): array
+    {
+        return [
+            'amount' => $this->_config->getRequestParameter('amount'),
+        ];
+    }
+
+    /**
+     * Validates the current request.
+     *
+     * @param Transaction $transaction
+     * @return void
+     */
+    public function validateRequest(Transaction $transaction): void
+    {
+        $requestParameters = $this->getRequestParameters();
+        $amount = $requestParameters['amount'];
+
+        if ($amount === null) {
+            return;
+        }
+
+        $amount = (float) $amount;
+
+        if (!$amount) {
+            throw new Exception($this->_i18n->translateString('text_generic_error'));
+        }
+
+        if ($amount < 0 || $amount > (float) $transaction->wdoxidee_ordertransactions__wdoxidee_amount->value) {
+            throw new Exception($this->_i18n->translateString('total_amount_not_in_range_text'));
+        }
+    }
+
+    /**
+     * Returns an array of processing actions to be displayed below the amount input field.
+     *
+     * @param Transaction $transaction
+     * @return array
+     */
+    protected function _getPostProcessingActions(Transaction $transaction): array
+    {
+        // TODO return only supported actions for the transaction
+        return [
```

Comment:
wouldn't it be easier to get that "info" from the payment method?
`Payment_Method` can have a default implementation and the specific ones can override if needed.

Reply:
That's also possible, yes.